libcontainer/process_linux.go

```diff
@@ -389,7 +431,47 @@ func (p *initProcess) start() (retErr error) {
 
 	ierr := parseSync(p.messageSockPair.parent, func(sync *syncT) error {
 		switch sync.Type {
+		case procSeccomp:
+			// receive seccomp-fd
+			pidfd, _, err := unix.Syscall(unix.SYS_PIDFD_OPEN, uintptr(childPid), 0, 0)
+			if err != 0 {
+				return fmt.Errorf("error performing SYS_PIDFD_OPEN syscall: %v", err)
+			}
+			defer unix.Close(int(pidfd))
+
+			seccompFd, _, err := unix.Syscall(unix.SYS_PIDFD_GETFD, pidfd, uintptr(sync.Fd), 0)
+			if err != 0 {
+				return fmt.Errorf("error performing SYS_PIDFD_GETFD syscall: %v", err)
+			}
+			defer unix.Close(int(seccompFd))
+
+			if p.config.Config.Seccomp.ListenerPath != "" {
+				s, err := p.container.currentOCIState()
+				if err != nil {
+					return err
+				}
+				// initProcessStartTime hasn't been set yet.
+				s.Pid = p.cmd.Process.Pid
+				s.Status = specs.StateCreating
+				seccompState := &specs.ContainerProcessState{
+					Version:   specs.Version,
+					FdIndexes: map[specs.FdIndexKey]int{specs.SeccompFdIndexKey: 0, specs.PidFdIndexKey: 1},
+					Pid:       s.Pid,
+					Metadata:  p.config.Config.Seccomp.ListenerMetadata,
+					State:     *s,
```

Comment:
Add a TODO in the code
```
// TODO: Add Annotations field
```

(or implement it)

Reply:
I don't get it. Do you mean `Annotations` in the `specs.State` structure? If you mean so, it's already present in the state returned by the `p.container.currentOCIState()`.

Reply:
Oh sorry, you're right